symfony/plugins/orangehrmAdminPlugin/Api/OrganizationAPI.php

```diff
@@ -0,0 +1,276 @@
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::POSITIVE)
+            ),
+        );
+    }
+
+    /**
+     * @return EndpointGetAllResult
+     * @throws Exception
+     */
+    public function getAll(): EndpointGetAllResult
+    {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     * @throws NotImplementedException
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function create(): EndpointCreateResult
+    {
+        // TODO:: Check data group permission
+        $orgInfo = $this->saveOrganizationInfo();
+
+        return new EndpointCreateResult(OrganizationModel::class, $orgInfo);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(self::PARAMETER_NAME),
+            new ParamRule(self::PARAMETER_TAX_ID),
+            new ParamRule(self::PARAMETER_REGISTRATION_NUMBER),
+            new ParamRule(self::PARAMETER_PHONE),
+            new ParamRule(self::PARAMETER_FAX),
+            new ParamRule(self::PARAMETER_EMAIL),
+            new ParamRule(self::PARAMETER_COUNTRY),
+            new ParamRule(self::PARAMETER_PROVINCE),
+            new ParamRule(self::PARAMETER_CITY),
+            new ParamRule(self::PARAMETER_ZIP_CODE),
+            new ParamRule(self::PARAMETER_STREET_1),
+            new ParamRule(self::PARAMETER_STREET_2),
+            new ParamRule(self::PARAMETER_NOTE),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function update(): EndpointUpdateResult
+    {
+        // TODO:: Check data group permission
+        $orgInfo = $this->saveOrganizationInfo();
+
+        return new EndpointUpdateResult(OrganizationModel::class, $orgInfo);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForUpdate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::POSITIVE)
+            ),
+            new ParamRule(self::PARAMETER_NAME),
+        );
+    }
+
+    /**
+     * @return Organization
+     * @throws DaoException
+     * @throws DaoException
+     */
+    public function saveOrganizationInfo(): Organization
+    {
+        // TODO:: Check data group permission
+        $name = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_NAME);
+        $taxId = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_TAX_ID);
```

Comment:
all below fields are optional right? shouldn't we use `$this->getRequestParams()->getStringOrNull`